# Tuberías en serie
## Comprobación de diseño
### **Ejemplo No.1**
> Una serie de 4 tuberias conecta dos tanques que forman parte del sistema de abastecimiento de agua de una finca dedicada a la piscicultura. La diferencia de altura entre los niveles de agua en los tanques es de 28.5 m. La primera tuberia de concreto tiene un diametro de 24 pulg, una longitud de 423 m, un coeficiente global de perdidas menores de 4.2, y un caudal lateral en su extremo final de 60 l/s, el cual alimenta el primer estanque de peces. La segunda tuberia en PVC, tiene un diametro de 20 pulg, una longitud de 174 m, un coeficiente global de perdidas menores de 3.4 y un caudal lateral de 74 l/s que llega al segundo estanque. La tercera tuberia, en PVC, tiene un diametro de 12 pulg, una longitud de 373 m, un coeficiente global de perdidas menores de 5.3 y un caudal lateral de de 60 l/s que alimenta el tercer estanque de peces. La ultima tuberia, en PVC, que llega al segundo tanque, tiene un diametro de 10 pulg, una longitud de 121 m, y un coeficiente global de perdidas menores de 7.5; dicha tuberia incluye una valvula de control totalmente abierta. Calcular el caudal que llega al segundo tanque. Como el agua esta a 14$^o$C:

> - $\rho$ = 999.2 kg/m$^3$
> - $\nu$ = 1.14 X 10$^{-6}$ m$^2$/s

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [1]:
import json
with open('designTest_js51.json', 'r') as f:
    data = json.load(f)
data

{'PT': 1,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.3,
 'mu': '',
 'nu': 1.14e-06,
 'E1': {'z': 28.5, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': -0.06,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.6096,
  'L': 423,
  'ks': 0.0003,
  'K': [4.2]},
 'P2': {'Qo': -0.074,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.508,
  'L': 174,
  'ks': 1.5e-06,
  'K': [3.4]},
 'P3': {'Qo': -0.06,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3048,
  'L': 373,
  'ks': 1.5e-06,
  'K': [5.3]},
 'P4': {'Qo': 0,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.254,
  'L': 121,
  'ks': 1.5e-06,
  'K': [7.5]}}

In [2]:
%run main.py designTest_js51.json


Proving the design of a serial pipe system

 
Iteration 1
 
          hf         V         Q        he         f             Re
P1  0.537550  0.928782  0.271078  0.184662  0.017620  496653.982051
P2  0.253310  1.041417  0.211078  0.187944  0.013379  464070.196537
P3  2.909985  1.878653  0.137078  0.953389  0.013219  502292.575828
P4  0.797900  1.521146  0.077078  0.884512  0.014202  338921.927144
 
Iteration 2
 
          hf         V         Q        he         f             Re
P1  0.948554  1.241823  0.362443  0.330118  0.017392  664048.315068
P2  0.488035  1.492196  0.302443  0.385862  0.012555  664943.396157
P3  7.402624  3.130816  0.228443  2.647841  0.012108  837081.241862
P4  3.321220  3.324259  0.168443  4.224274  0.012378  740668.326385
 
Iteration 3
 
          hf         V         Q        he         f             Re
P1  1.113620  1.347662  0.393333  0.388787  0.017337  720644.303392
P2  0.582940  1.644604  0.333333  0.468708  0.012346  732858.582146
P3  9.341460  3.554171 

### **Ejemplo No.2**
> La figura 6.7 ilustra un sistema de 3 tuberias conectadas en serie, cuyas caracteristicas geometricas estan definidas en la tabla 6.2. Determinar el caudal que transporta el sistema.

> $\rho$ = 999.3 kg/m$^3$

> $\nu$ = 1.0 X 10$^{-3}$ m$^2$/s
 
![alternatvie text](./figs/fcd62.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [3]:
import json
with open('designTest_cd62.json', 'r') as f:
    data = json.load(f)
data

{'PT': 1,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.3,
 'mu': '',
 'nu': 1e-06,
 'E1': {'z': 10, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': 0,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3,
  'L': 300,
  'ks': 0.00025,
  'K': [0]},
 'P2': {'Qo': 0,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.5,
  'L': 300,
  'ks': 0.00025,
  'K': [0]},
 'P3': {'Qo': 0,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.3,
  'L': 600,
  'ks': 0.00025,
  'K': [0]}}

In [1]:
%run main.py designTest_cd62.json


Proving the design of a serial pipe system

 
Iteration 1
 
          hf         V         Q   he         f             Re
P1  3.249116  1.811702  0.128062  0.0  0.019422  543510.549937
P2  0.235523  0.652213  0.128062  0.0  0.018105  326106.329962
P3  6.498232  1.811702  0.128062  0.0  0.019422  543510.549937
 
Iteration 2
 
          hf         V         Q   he         f             Re
P1  3.254681  1.813277  0.128173  0.0  0.019421  543983.032326
P2  0.235919  0.652780  0.128173  0.0  0.018104  326389.819395
P3  6.509363  1.813277  0.128173  0.0  0.019421  543983.032326
 
Iteration 3
 
          hf         V         Q   he         f             Re
P1  3.254693  1.813280  0.128173  0.0  0.019421  543984.048113
P2  0.235920  0.652781  0.128173  0.0  0.018104  326390.428868
P3  6.509386  1.813280  0.128173  0.0  0.019421  543984.048113


## Calculo de la potencia requerida
### **Ejemplo No.1**
> Como parte del sistema de riego de una finca se utilizan dos tuberias en serie para conectar la bocatoma con un tanque de almacenamiento. La diferencia de nivel entre estas dos estructuras es de 31.7 m, estando la bocatoma por debajo del tanque. El caudal que debe llegar al tanque es de 87 l/s. La primera tuberia, en acero, tiene un diametro de 8 pulg, una longitud de 184 m y un coeficiente global de perdidas menores de 7.1. Al final de esta debe extraerse un caudal de 94 l/s con el fin de regar la parte baja de la finca. La segunda tuberia en PVC, tiene una longitud de 393 m, un diametro de 6 pulg, y un coeficiente global de perdidas menores de 11.2, el cual incluye la valvula de control. Calcular la potencia de la bomba requerida para realizar el trabajo. El fluido es agua a 15 $^o$C:

> - $\rho$ = 999.3 kg/m$^3$

> - $\nu$ = 1.14 X 10$^{-6}$ m$^2$/s

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [2]:
import json
with open('systemPower_js52.json', 'r') as f:
    data = json.load(f)
data

{'PT': 2,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.3,
 'mu': '',
 'nu': 1.14e-06,
 'E1': {'z': 0, 'p': 0, 'v': 0},
 'E2': {'z': 31.7, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': -0.094,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 0.75},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.2032,
  'L': 184,
  'ks': 4.6e-05,
  'K': [7.1]},
 'P2': {'Qo': -0.087,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.1524,
  'L': 393,
  'ks': 1.5e-06,
  'K': [11.2]}}

In [3]:
%run main.py systemPower_js52.json


Estimating the power in serial pipes

           hf         V      Q         he         f             Re
P1  21.516123  5.581376  0.181  11.273063  0.014965  994855.805347
P2  38.156354  4.769353  0.087  12.984879  0.012763  637587.145968

hp =   115.63 (m)
P = 273561.08 (W)


### **Ejemplo No.2**
> Calcule la altura H del tanque mostrado en la figura teniendo en cuenta que el caudal que debe llegar a la piscina es de 54.3 l/s. El material de las tuberias es hierro galvanizado.

> - $\rho$ = 999.1 kg/m$^3$
> - $\nu$ = 1.141 X 10$^{-6}$ m$^2$/s

![alternatvie text](./figs/fjs55.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [4]:
import json
with open('systemPower_js55.json', 'r') as f:
    data = json.load(f)
data

{'PT': 2,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': '',
 'nu': 1.141e-06,
 'E1': {'z': 0, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': -0.0303,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.254,
  'L': 393,
  'ks': 0.00015,
  'K': [12.3]},
 'P2': {'Qo': -0.0288,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.2032,
  'L': 125,
  'ks': 0.00015,
  'K': [6.9]},
 'P3': {'Qo': -0.0543,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': 0.1524,
  'L': 230,
  'ks': 0.00015,
  'K': [7.8]}}

In [5]:
%run main.py systemPower_js55.json


Estimating the power in serial pipes

           hf         V       Q        he         f             Re
P1   7.190890  2.237978  0.1134  3.139913  0.018206  498200.102041
P2   3.925567  2.562499  0.0831  2.309285  0.019067  456353.929449
P3  13.845022  2.976734  0.0543  3.522700  0.020313  397593.555862

hp =    33.93 (m)
P = 37715.35 (W)


## Diseno de tuberias
### **Ejemplo No.1**
> Una de las tuberias principales de un sistema de riego localizado de alta frequencia debe conectar la estacion de fertirrigacion con tres modulos de riego que operan en forma simultanea. Los caudales consumidos en estos son: modulo A, 45.1 l/s; modulo B, 39.0 l/s; modulo C, 73.2 l/s. En la estacion de bombeo se cuenta con una bomba de 65 kW con una eficiencia del 85%. La tuberia 1, que va de la estacion hasta el modulo A, tiene una longitud de 350 m y un coeficiente global de perdidas menores de 7.9. La tuberia 2, que une el modulo A con el modulo B, tiene una longitud de 123 m y un coeficiente global de perdidas menores de 3.3. Finalmente, la tuberia 3, que va del modulo B al modulo C, tiene una longitud de 174 m y un coeficiente global de perdidas menores de 3.5. Todo el sistema se encuentra en un terreno approximadamente horizontal y el agua bombeada se encuentra a 15 $^o$C. Diseñar las tres tuberias si el material disponible es PVC.

> - $\rho$ = 999.1 kg/m$^3$

> - $\nu$ = 1.14 X 10$^{-6}$ m$^2$/s

![alternatvie text](./figs/fjs53.jpeg)

> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [6]:
import json
with open('pipeDesign_js53.json', 'r') as f:
    data = json.load(f)
data

{'PT': 3,
 'US': 'IS',
 'IM': 'fp',
 'rho': 999.1,
 'mu': '',
 'nu': 1.14e-06,
 'E1': {'z': 0, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': -0.0451,
  'Qi': 0,
  'Pu': {'P': 65000.0, 'h': '', 'ef': 0.85},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 350,
  'ks': 1.5e-06,
  'K': [7.9]},
 'P2': {'Qo': -0.039,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 123,
  'ks': 1.5e-06,
  'K': [3.3]},
 'P3': {'Qo': -0.0732,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 174,
  'ks': 1.5e-06,
  'K': [3.5]}}

In [7]:
%run main.py pipeDesign_js53.json


Design of a serial pipe system

 
Iteration 1
 
    D"        hf         V       Q        he         f             Re
P1  10  8.475585  3.104355  0.1573  3.880350  0.012523  691672.124557
P2   8  4.725136  3.459836  0.1122  2.013381  0.012794  616700.670497
P3   8  3.065494  2.257219  0.0732  0.908901  0.013786  402339.474870

Balance of energy at the end:    12.77
 
Iteration 2
 
    D"         hf         V       Q        he         f             Re
P1  10   8.475585  3.104355  0.1573  3.880350  0.012523  691672.124557
P2   8   4.725136  3.459836  0.1122  2.013381  0.012794  616700.670497
P3   6  12.314152  4.012835  0.0732  2.872576  0.013141  536452.633160

Balance of energy at the end:     1.56
 
Iteration 3
 
    D"        hf         V       Q        he         f             Re
P1  12  3.510034  2.155802  0.1573  1.871311  0.012904  576393.437131
P2  10  1.606595  2.214295  0.1122  0.824681  0.013276  493360.536398
P3   8  3.065494  2.257219  0.0732  0.908901  0.013786  402339.47

### **Ejemplo No.2**
> En una planta de tratamiento de agua potable es necesario repartir el agua cruda a tres tanques floculadores, tal como se muestra en la figura. Calcule el diametro de cada una de las tres tuberias si su material es hierro galvanizado.

> - $\rho$ = 999.1 kg/m$^3$

> - $\nu$ = 1.14 X 10$^{-6}$ m$^2$/s

![alternatvie text](./figs/fjs59.jpeg)


> *Solucion:*

> Siguiendo el procedimiento descrito, se tiene:

> La imformacion de entrada al programa es:

In [8]:
import json
with open('pipeDesign_js59.json', 'r') as f:
    data = json.load(f)
data

{'PT': 3,
 'US': 'IS',
 'IM': 'nr',
 'rho': 999.1,
 'mu': '',
 'nu': 1.14e-06,
 'E1': {'z': 19.3, 'p': 0, 'v': 0},
 'E2': {'z': 0, 'p': 0, 'v': 0},
 'Ki': 0.0,
 'Ko': 0.0,
 'P1': {'Qo': -0.16,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 72,
  'ks': 0.00015,
  'K': [6.4]},
 'P2': {'Qo': -0.16,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 45,
  'ks': 0.00015,
  'K': [3.6]},
 'P3': {'Qo': -0.16,
  'Qi': 0,
  'Pu': {'P': '', 'h': '', 'ef': 1},
  'Tu': {'P': '', 'h': '', 'ef': 1},
  'D': '',
  'L': 45,
  'ks': 0.00015,
  'K': [3.6]}}

In [9]:
%run main.py pipeDesign_js59.json


Design of a serial pipe system

 
Iteration 1
 
    D"        hf         V     Q        he         f            Re
P1  16  1.990951  3.700360  0.48  4.466516  0.016102  1.319146e+06
P2  14  1.114879  3.222082  0.32  1.904920  0.016650  1.005063e+06
P3  10  1.617838  3.157640  0.16  1.829485  0.017969  7.035444e+05

Balance of energy at the end:     6.38
 
Iteration 2
 
    D"        hf         V     Q        he         f            Re
P1  16  1.990951  3.700360  0.48  4.466516  0.016102  1.319146e+06
P2  12  2.472174  4.385612  0.32  3.529099  0.017081  1.172574e+06
P3  10  1.617838  3.157640  0.16  1.829485  0.017969  7.035444e+05

Balance of energy at the end:     3.39
 
Iteration 3
 
    D"        hf         V     Q        he         f            Re
P1  16  1.990951  3.700360  0.48  4.466516  0.016102  1.319146e+06
P2  14  1.114879  3.222082  0.32  1.904920  0.016650  1.005063e+06
P3  10  1.617838  3.157640  0.16  1.829485  0.017969  7.035444e+05

Balance of energy at the end:     